src/app/pages/onboarding/set-password/components/password-strength-indicator.tsx

```diff
@@ -1,50 +1,44 @@
 import { useMemo } from 'react';
 
-import { Box, HStack } from 'leather-styles/jsx';
-
 import { createNullArrayOfLength } from '@app/common/utils';
 import { ValidatedPassword } from '@app/common/validation/validate-password';
 
+import { PasswordStrengthBars } from './password-bars';
 import { defaultColor } from './password-field.utils';
 
 function fillArray(amount: number) {
-  return (item: (i: number) => React.JSX.Element) =>
-    createNullArrayOfLength(amount).map((_, i) => item(i));
+  return (item: (i: number) => string) => createNullArrayOfLength(amount).map((_, i) => item(i));
 }
 
 interface PasswordStrengthIndicatorProps {
+  password: string;
   strengthColor: string;
   strengthResult: ValidatedPassword;
 }
-export function PasswordStrengthIndicator(props: PasswordStrengthIndicatorProps) {
-  const { strengthColor, strengthResult } = props;
-
+export function PasswordStrengthIndicator({
+  password,
```

Comment:
I'm passing `password` now also so that if users delete what they have typed it clears the bars